CMakePresets.json

```diff
@@ -0,0 +1,314 @@
+            "inherits": ["common", "build-debug"]
+        },
+        {
+            "name": "debug-ninja-sscache",
+            "configurePreset": "debug-ninja-sscache",
+            "inherits": ["common", "build-debug"]
+        },
+        {
+            "name": "debug-qt5",
+            "configurePreset": "debug-qt5",
+            "inherits": ["common", "build-debug"]
+        },
+        {
+            "name": "debug-qt5-ninja-sscache",
+            "configurePreset": "debug-qt5-ninja-sscache",
+            "inherits": ["common", "build-debug"]
+        },
+        {
+            "name": "vcpkg-debug",
+            "configurePreset": "vcpkg-debug",
+            "inherits": ["common", "build-debug"]
+        },
+        {
+            "name": "vcpkg-debug-ninja-sscache",
+            "configurePreset": "vcpkg-debug-ninja-sscache",
+            "inherits": ["common", "build-debug"]
+        },
+        {
+            "name": "vcpkg-debug-qt5",
+            "configurePreset": "vcpkg-debug-qt5",
+            "inherits": ["common", "build-debug"]
+        },
+        {
+            "name": "vcpkg-debug-qt5-ninja-sscache",
+            "configurePreset": "vcpkg-debug-qt5-ninja-sscache",
+            "inherits": ["common", "build-debug"]
+        },
+        {
+            "name": "release",
+            "configurePreset": "release",
+            "inherits": ["common", "build-release"]
+        },
+        {
+            "name": "release-ninja-sscache",
+            "configurePreset": "release-ninja-sscache",
+            "inherits": ["common", "build-release"]
+        },
+        {
+            "name": "release-qt5",
+            "configurePreset": "release-qt5",
+            "inherits": ["common", "build-release"]
+        },
+        {
+            "name": "release-qt5-ninja-sscache",
+            "configurePreset": "release-qt5-ninja-sscache",
+            "inherits": ["common", "build-release"]
+        },
+        {
+            "name": "vcpkg-release",
+            "configurePreset": "vcpkg-release",
+            "inherits": ["common", "build-release"]
+        },
+        {
+            "name": "vcpkg-release-ninja-sscache",
+            "configurePreset": "vcpkg-release-ninja-sscache",
+            "inherits": ["common", "build-release"]
+        },
+        {
+            "name": "vcpkg-release-qt5",
+            "configurePreset": "vcpkg-release-qt5-ninja-sscache",
+            "inherits": ["common", "build-release"]
+        },
+        {
+            "name": "vcpkg-release-qt5-ninja-sscache",
+            "configurePreset": "vcpkg-release-qt5-ninja-sscache",
+            "inherits": ["common", "build-release"]
+        }
+    ],
+    "testPresets": [
+        {
+            "name": "build-debug",
+            "configuration": "Debug",
+            "hidden": true
+        },
+        {
+            "name": "build-release",
+            "configuration": "Release",
+            "hidden": true
+        },
+        {
```

Comment:
If we do use this we likely also want to set the parallel / execution jobs to something larger than 1

```
"execution": {
    "jobs": 2
},
```